pkg/client/client.go

```diff
@@ -1843,17 +1852,38 @@ func WithLastError(e *error) func(o *pollOptions) {
 // Poll is a wrapper around wait.PollUntilContextTimeout that allows adding the passed lastError into
 // the final error if set by the condition, adding more context to the "context deadline exceeded" error.
 func Poll(ctx context.Context, condition wait.ConditionWithContextFunc, options ...func(o *pollOptions)) error {
-	opts := pollOptions{timeout: 5 * time.Minute, interval: time.Second}
+	opts := pollOptions{
+		timeout:  5 * time.Minute,
+		interval: time.Second,
+	}
 	for _, o := range options {
 		o(&opts)
 	}
 
-	if err := wait.PollUntilContextTimeout(ctx, opts.interval, opts.timeout, false, condition); err != nil {
+	var (
+		conditionErr error
```

Comment:
why do we need this `conditionErr`?
A unit test would be great to show when `Poll` couldn't catch such error.

Reply:
we need it at L1881. Eventually I think that we can rid of the WithLastError() construct too.

Reply:
IMHO a simpler contract is to declare that the function is retried is retried until no error is returned. From what I can tell, there's zero occurrences where the condition function returns an error directly (e.g. it's always `return false, nil` or `return true, nil`).

Reply:
Yep, but I still think that catching the `condition` error (via `conditionErr`) should be done in a separate PR. AFAIU, this doesn't help with anything regarding OCPBUGS-17506, but I might be missing something. A unit test showing the current error-catching logic's shortcomings would be helpful.

We can leave this for another PR and also see if we can completely remove the `lastError` logic.

/lgtm 
Thanks for this!

/hold
Please unhold if/when needed.